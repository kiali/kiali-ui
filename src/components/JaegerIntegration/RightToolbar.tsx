import * as React from 'react';
import { ToolbarRightContent, Button, Icon } from 'patternfly-react';

interface RightToolbarProps {
  disabled: boolean;
  onSubmit: () => void;
}

const RightToolbar = (props: RightToolbarProps) => (
  <ToolbarRightContent>
    <Button
      bsStyle={'primary'}
      bsSize={'lg'}
      style={{ fontSize: '15px' }}
      title={'Search'}
      onClick={() => props.onSubmit()}
      disabled={props.disabled}
    >
      <Icon type="pf" name="search" /> Search
    </Button>
  </ToolbarRightContent>
);
export default RightToolbar;
