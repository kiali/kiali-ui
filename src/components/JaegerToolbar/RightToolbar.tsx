import * as React from 'react';
import { ToolbarRightContent, Button, Icon } from 'patternfly-react';

interface RightToolbarProps {
  disabled: boolean;
  graph: boolean;
  minimap: boolean;
  summary: boolean;
  onGraphClick: (state: boolean) => void;
  onSummaryClick: (state: boolean) => void;
  onMinimapClick: (state: boolean) => void;
  onSubmit: () => void;
}

export class RightToolbar extends React.PureComponent<RightToolbarProps, {}> {
  static active = { color: '#0088ce' };

  constructor(props: RightToolbarProps) {
    super(props);
  }

  render() {
    const { disabled, graph, minimap, summary, onGraphClick, onSummaryClick, onMinimapClick, onSubmit } = this.props;
    return (
      <ToolbarRightContent>
        <Button title={'Graph'} style={graph ? RightToolbar.active : undefined} onClick={() => onGraphClick(graph)}>
          <Icon type="fa" name="th" />
        </Button>
        <Button
          title={'Minimap'}
          style={minimap ? RightToolbar.active : undefined}
          onClick={() => onMinimapClick(minimap)}
        >
          <Icon type="fa" name="map" />
        </Button>
        <Button
          title={'Summary'}
          style={summary ? RightToolbar.active : undefined}
          onClick={() => onSummaryClick(summary)}
        >
          <Icon type="fa" name="info" />
        </Button>
        <Button
          bsStyle={'link'}
          title={'Search'}
          style={{ borderLeft: '1px solid #d1d1d1', marginLeft: '10px' }}
          onClick={onSubmit}
          disabled={disabled}
        >
          <Icon type="pf" name="search" />
        </Button>
      </ToolbarRightContent>
    );
  }
}

export default RightToolbar;
