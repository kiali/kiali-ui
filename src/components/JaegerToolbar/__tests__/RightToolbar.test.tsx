import * as React from 'react';
import { shallow } from 'enzyme';
import { RightToolbar } from '../RightToolbar';

const active = { color: '#0088ce' };
describe('RightToolbar', () => {
  let wrapper, onGraphClick, onSummaryClick, onMinimapClick, onSubmit;
  beforeEach(() => {
    onGraphClick = jest.fn();
    onMinimapClick = jest.fn();
    onSummaryClick = jest.fn();
    onSubmit = jest.fn();
    const props = {
      disabled: false,
      graph: false,
      minimap: false,
      summary: false,
      onGraphClick: onGraphClick,
      onSummaryClick: onSummaryClick,
      onMinimapClick: onMinimapClick,
      onSubmit: onSubmit
    };
    wrapper = shallow(<RightToolbar {...props} />);
  });

  it('renders RightToolbar correctly', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  describe('RightToolbar should have buttons with options', () => {
    it('RightToolbar have Graph button', () => {
      let buttonProps = wrapper.find({ title: 'Graph' }).props();
      expect(buttonProps).toBeDefined();
      expect(buttonProps['style']).toBeUndefined();
      expect(buttonProps['onClick']).toBeDefined();
      wrapper.setProps({ graph: true });
      buttonProps = wrapper.find({ title: 'Graph' }).props();
      expect(buttonProps['style']).toEqual(active);
    });

    it('RightToolbar have Minimap button', () => {
      let buttonProps = wrapper.find({ title: 'Minimap' }).props();
      expect(buttonProps).toBeDefined();
      expect(buttonProps['style']).toBeUndefined();
      expect(buttonProps['onClick']).toBeDefined();
      wrapper.setProps({ minimap: true });
      buttonProps = wrapper.find({ title: 'Minimap' }).props();
      expect(buttonProps['style']).toEqual(active);
    });

    it('RightToolbar have Summary button', () => {
      let buttonProps = wrapper.find({ title: 'Summary' }).props();
      expect(buttonProps).toBeDefined();
      expect(buttonProps['style']).toBeUndefined();
      expect(buttonProps['onClick']).toBeDefined();
      wrapper.setProps({ summary: true });
      buttonProps = wrapper.find({ title: 'Summary' }).props();
      expect(buttonProps['style']).toEqual(active);
    });

    it('RightToolbar have Search button', () => {
      let buttonProps = wrapper.find({ title: 'Search' }).props();
      expect(buttonProps).toBeDefined();
      expect(buttonProps['onClick']).toBeDefined();
      expect(buttonProps['style']).toEqual({ borderLeft: '1px solid #d1d1d1', marginLeft: '10px' });
      wrapper.setProps({ disabled: true });
      buttonProps = wrapper.find({ title: 'Search' }).props();
      expect(buttonProps['disabled']).toBeTruthy();
    });
  });
});
