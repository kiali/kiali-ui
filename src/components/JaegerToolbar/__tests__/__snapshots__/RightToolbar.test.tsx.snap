// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightToolbar renders RightToolbar correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RightToolbar
    disabled={false}
    graph={false}
    minimap={false}
    onGraphClick={[MockFunction]}
    onMinimapClick={[MockFunction]}
    onSubmit={[MockFunction]}
    onSummaryClick={[MockFunction]}
    summary={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          style={undefined}
          title="Graph"
        >
          <Icon
            name="th"
            type="fa"
          />
        </Button>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          style={undefined}
          title="Minimap"
        >
          <Icon
            name="map"
            type="fa"
          />
        </Button>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          style={undefined}
          title="Summary"
        >
          <Icon
            name="info"
            type="fa"
          />
        </Button>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
          onClick={[MockFunction]}
          style={
            Object {
              "borderLeft": "1px solid #d1d1d1",
              "marginLeft": "10px",
            }
          }
          title="Search"
        >
          <Icon
            name="search"
            type="pf"
          />
        </Button>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": <Icon
            name="th"
            type="fa"
          />,
          "disabled": false,
          "onClick": [Function],
          "style": undefined,
          "title": "Graph",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "th",
            "type": "fa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": <Icon
            name="map"
            type="fa"
          />,
          "disabled": false,
          "onClick": [Function],
          "style": undefined,
          "title": "Minimap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "map",
            "type": "fa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": <Icon
            name="info"
            type="fa"
          />,
          "disabled": false,
          "onClick": [Function],
          "style": undefined,
          "title": "Summary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "info",
            "type": "fa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "link",
          "children": <Icon
            name="search"
            type="pf"
          />,
          "disabled": false,
          "onClick": [MockFunction],
          "style": Object {
            "borderLeft": "1px solid #d1d1d1",
            "marginLeft": "10px",
          },
          "title": "Search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "search",
            "type": "pf",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
            style={undefined}
            title="Graph"
          >
            <Icon
              name="th"
              type="fa"
            />
          </Button>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
            style={undefined}
            title="Minimap"
          >
            <Icon
              name="map"
              type="fa"
            />
          </Button>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function]}
            style={undefined}
            title="Summary"
          >
            <Icon
              name="info"
              type="fa"
            />
          </Button>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="link"
            disabled={false}
            onClick={[MockFunction]}
            style={
              Object {
                "borderLeft": "1px solid #d1d1d1",
                "marginLeft": "10px",
              }
            }
            title="Search"
          >
            <Icon
              name="search"
              type="pf"
            />
          </Button>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": <Icon
              name="th"
              type="fa"
            />,
            "disabled": false,
            "onClick": [Function],
            "style": undefined,
            "title": "Graph",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "th",
              "type": "fa",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": <Icon
              name="map"
              type="fa"
            />,
            "disabled": false,
            "onClick": [Function],
            "style": undefined,
            "title": "Minimap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "map",
              "type": "fa",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": <Icon
              name="info"
              type="fa"
            />,
            "disabled": false,
            "onClick": [Function],
            "style": undefined,
            "title": "Summary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "info",
              "type": "fa",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "link",
            "children": <Icon
              name="search"
              type="pf"
            />,
            "disabled": false,
            "onClick": [MockFunction],
            "style": Object {
              "borderLeft": "1px solid #d1d1d1",
              "marginLeft": "10px",
            },
            "title": "Search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "search",
              "type": "pf",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
